tests/integration/test_custom_application_source_version.py

```diff
@@ -174,32 +175,32 @@ def test_get_or_create(
             f"customApplicationSources/{custom_application_source}/versions/{env_ver_id_3}"
         ).json()["items"]
     ]
-    env_ver_id_4 = get_or_create_custom_application_source_version(
+    env_ver_id_4 = get_or_create_custom_application_source_version_from_previous(
```

Comment:
i still am not tracking why we can't create this same experience (e.g. test will pass) using a single `get_or_create_custom_application_source_version` ?

Reply:
Somehow I have to differentiate between creating a new thing from scratch vs building on top of an existing thing

Reply:
Surely we have to have something that reflects this:

![image](https://github.com/datarobot-community/datarobotx-idp/assets/6739699/52fdb346-8ee8-4c1c-907c-ffa1de23b0b3)

Reply:
which is the same as this (in custom model workshop)
![image](https://github.com/datarobot-community/datarobotx-idp/assets/6739699/15154fec-ae57-45c0-8d0b-ae4290215d42)

Reply:
the change I introduced is that none of our functions allow for modifying an existing version in place. it's always a new version, or retrieve existing. No more version content changes but version id stays the same